pkg/cmd/pulumi/state_move.go

```diff
@@ -141,6 +142,9 @@ func (cmd *stateMoveCmd) Run(
 	for _, res := range sourceSnapshot.Resources {
 		matchedArg := resourceMatches(res, args)
 		if matchedArg != "" {
+			if strings.HasPrefix(string(res.Type), "pulumi:providers:") {
```

Comment:
It kinda surprises me that this is the best way we have to recognise a provider.